src/NBitcoin/StandardScriptTemplate.cs

```diff
@@ -297,6 +298,136 @@ public Script GenerateScriptSig(IEnumerable<TransactionSignature> signatures)
         }
     }
 
+    public class PayToFederationTemplate : ScriptTemplate
+    {
+        private static readonly PayToFederationTemplate _Instance = new PayToFederationTemplate();
+
+        public static PayToFederationTemplate Instance
+        {
+            get
+            {
+                return _Instance;
+            }
+        }
+
+        public Script GenerateScriptPubKey(IFederationId federationId)
+        {
+            var ops = new List<Op>();
+            ops.Add(Op.GetPushOp(federationId.ToBytes()));
+            ops.Add(OpcodeType.OP_FEDERATION);
+            ops.Add(OpcodeType.OP_CHECKMULTISIG);
+            return new Script(ops);
+        }
+
+        protected override bool CheckScriptPubKeyCore(Script scriptPubKey, Op[] scriptPubKeyOps)
+        {
+            Op[] ops = scriptPubKeyOps;
+            if (ops.Length < 3)
+                return false;
+
+            byte[] federationId = ops[0].PushData;
+
+            // TODO: Validate the federation id.
```

Comment:
I think this is advisable

Reply:
Actually, this probably doesn't matter all that much - when the OP_FEDERATION opcode gets evaluated the specific federation details get looked up anyway (if they exist). So I think for forwards compatibility we don't want to restrict what constitutes a valid scriptPubKey yet

Reply:
Ok.